### PR TITLE
Added gorm and birthdate check :)

### PR DESCRIPTION
**Pull request by braydencw1**
Added gorm and birthdate check :)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -41,29 +45,38 @@ func main() {
...
 	// Keep the bot running
 	log.Println("Bot is now running. Press Ctrl+C to exit.")
+	dbUsername := os.Getenv("DB_USER")
+	dbHost := os.Getenv("DB_HOST")
+	dbDB := os.Getenv("DB_DB")
+	dbPassword := os.Getenv("DB_PASS")
+	dbPort := os.Getenv("DB_PORT")
+	dsn := fmt.Sprintf("user=%s password=%s host=%s port=%s dbname=%s sslmode=disable", dbUsername, dbPassword, dbHost, dbPort, dbDB)
+	db.OpenDatabase(dsn)
+	//	birthdateCheck(discord)
```
Comment (crisleiva) [inline main.go]: Here

File: main.go [hunk 2]
```diff
@@ -5,11 +5,15 @@ import (
...
 	"os"
 	"strings"
+	"time"
+	"venova/db"
 
 	"github.com/bwmarrin/discordgo"
 	"github.com/joho/godotenv"
 )
 
+// var tcTrainingId string = "1026643518942355476"
```
Comment (crisleiva) [inline main.go]: here

File: main.go [hunk 3]
```diff
@@ -72,54 +85,73 @@ func addGriefer(s *discordgo.Session, m *discordgo.MessageCreate) {
...
-	if m.ChannelID != channelId {
+func handleVoiceStateUpdate(sess *discordgo.Session, mess *discordgo.VoiceStateUpdate) {
+	if mess.ChannelID != channelId {
 		return
 	}
 
-	if m.VoiceState.UserID == morthisId {
-		s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("Hello gaylord <@%s>", morthisId))
+	if mess.VoiceState.UserID == morthisId {
+		sess.ChannelMessageSend(mess.ChannelID, fmt.Sprintf("Hello gaylord <@%s>", morthisId))
```
Comment (crisleiva) [inline main.go]: 😆

File: db/db.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	return nil
+}
+
+type User struct {
+	ID         int `gorm:"primaryKey"`
+	Disc_id    int
+	First_name string
+	Last_name  string
+	Dob        time.Time
+}
```
Comment (b1naryth1ef) [inline db/db.go]:
```suggestion
type User struct {
	Id         int `gorm:"primaryKey"`
	DiscordId    int
	FirstName string
	LastName  string
	Dob        time.Time
}
```

if gorm complains about fields (or they come back empty) you need to use a struct field tag to override the column name (which is explained in the gorm docs [here](https://gorm.io/docs/conventions.html#Column-Name))

File: db/db.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+func GetBirthdays(dateToCheck time.Time) ([]int, error) {
+	if db == nil {
```
Comment (b1naryth1ef) [inline db/db.go]: Remove this check, we won't need it.

File: main.go [hunk 6]
```diff
@@ -41,29 +45,38 @@ func main() {
 	discord.AddHandler(handleVoiceStateUpdate)
 	// Keep the bot running
 	log.Println("Bot is now running. Press Ctrl+C to exit.")
+	dbUsername := os.Getenv("DB_USER")
+	dbHost := os.Getenv("DB_HOST")
+	dbDB := os.Getenv("DB_DB")
+	dbPassword := os.Getenv("DB_PASS")
+	dbPort := os.Getenv("DB_PORT")
+	dsn := fmt.Sprintf("user=%s password=%s host=%s port=%s dbname=%s sslmode=disable", dbUsername, dbPassword, dbHost, dbPort, dbDB)
+	db.OpenDatabase(dsn)
```
Comment (b1naryth1ef) [inline main.go]: Check the error result here and panic if its not nil so we don't startup in an invalid state.

**Review by crisleiva — APPROVED** [review]

File: db/db.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+		fmt.Println("Error: ", res.Error)
+		return nil, res.Error
+	}
+	var discIds []int
+	for _, user := range users {
+		log.Printf("Today's birthdays are: %v", user)
+		discIds = append(discIds, user.DiscordId)
+	}
+	return discIds, nil
+}
```
Comment (b1naryth1ef) [inline db/db.go]: Lets move this to another file, `db/user.go`.

File: main.go (hunk 2 above)
Comment (b1naryth1ef) [inline main.go]: DELETE THIS MOFO.

File: main.go [hunk 8]
```diff
@@ -41,29 +45,41 @@ func main() {
...
+	dbHost := os.Getenv("DB_HOST")
+	dbDB := os.Getenv("DB_DB")
+	dbPassword := os.Getenv("DB_PASS")
+	dbPort := os.Getenv("DB_PORT")
+	dsn := fmt.Sprintf("user=%s password=%s host=%s port=%s dbname=%s sslmode=disable", dbUsername, dbPassword, dbHost, dbPort, dbDB)
+	err = db.OpenDatabase(dsn)
+	if err != nil {
+		log.Panicf("Database connection is rough, to say the least: %v", err)
+	}
+	//	birthdateCheck(discord)
```
Comment (b1naryth1ef) [inline main.go]: DELETE THIS MOFO

**Review by b1naryth1ef — APPROVED** [review]

**Pull request merged**